CMake inserting unwanted quote characters

## Expected Behavior

MPI Configurations should work fine.

## Actual Behavior

When the MPI library has more than one flag defined, the MPI flags are added to CXX_FLAGS with quotes around them.  For example, the openmpi build I am using defines `MPI_CXX_COMPILE_OPTIONS=-fexceptions;-pthread`. When that gets added to the CXX_FLAGS, it is added with quotes (and fails to compile).  For example, `plugins/parcelport/mpi/CMakeFiles/parcelport_mpi.dir/flags.make` looks like this:

```
CXX_FLAGS = -fPIC -O3 -DNDEBUG -fPIC   -std=c++17 "-fexceptions -pthread" -pthread -Wall -Wextra -Wno-unused-local-typedefs -Wno-strict-aliasing -Wno-sign-promo -Wno-attributes -Wno-cast-align -Wno-unused-parameter -Wformat=2 -Wno-format-nonliteral -Winit-self -Wdouble-promotion -Wcast-qual -Wcast-align -Werror=trampolines -Werror=parentheses -Werror=reorder -Werror=return-type -Werror=sequence-point -Werror=uninitialized -Werror=format -Werror=missing-braces -Werror=sign-compare -fdiagnostics-show-option -Werror=vla -Wno-unused-but-set-parameter -Wno-unused-but-set-variable -Wno-maybe-uninitialized -Wno-sync-nand -Wno-delete-non-virtual-dtor -fvisibility=hidden
```

Which generates this compiler error:

```
g++: error: unrecognized command line option '-fexceptions -pthread'; did you mean '-fexceptions'?
make[3]: *** [plugins/parcelport/mpi/CMakeFiles/parcelport_mpi.dir/mpi_environment.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
```

## Steps to Reproduce the Problem

  1.  Configure HPX with MPI enabled, ideally one with more than one compiler flag.  For example, in openmpi, you can change the pkgconfig files to have: `Cflags: -I${includedir}     -fexceptions -pthread`
  1.  After configuring HPX, see that the flags have quotes around them, i.e. in `src/CMakeFiles/hpx.dir/flags.make` or `plugins/parcelport/mpi/CMakeFiles/parcelport_mpi.dir/flags.make`

## Specifications

... Please describe your environment

  - HPX Version:  current master
  - Platform (compiler, OS):  linux, x86_64, gcc 7.3

--- a/cmake/HPX_SetupMPI.cmake
+++ b/cmake/HPX_SetupMPI.cmake
@@ -35,14 +35,11 @@ macro(setup_mpi)
     if(MPI_EXTRA_LIBRARY)
       target_link_libraries(Mpi::mpi INTERFACE ${MPI_EXTRA_LIBRARY})
     endif()
+    separate_arguments(MPI_CXX_COMPILE_FLAGS)
     target_compile_options(Mpi::mpi INTERFACE ${MPI_CXX_COMPILE_FLAGS})
     target_compile_definitions(
       Mpi::mpi INTERFACE ${MPI_CXX_COMPILE_DEFINITIONS}
     )
-
-    if(MPI_CXX_LINK_FLAGS)
-      # hpx_add_link_flag_if_available(${MPI_CXX_LINK_FLAGS})
-    endif()
 
     hpx_info("MPI version: " ${MPI_C_VERSION})
   endif()

